input/signdescriptor.go

```diff
@@ -307,3 +315,29 @@ func ReadSignDescriptor(r io.Reader, sd *SignDescriptor) error {
 
 	return nil
 }
+
+// MaybeEnrichPsbt populates a PSBT pakcet's partial outputs with extra data
+// provided in a sign descriptor, if any.
+func MaybeEnrichPsbt(packet *psbt.Packet, outputs []psbt.POutput) error {
+	if len(outputs) == 0 {
+		return nil
+	}
+
+	if len(packet.Outputs) != len(outputs) {
+		return fmt.Errorf("partial output count mismatch: need %d, "+
+			"got %d", len(packet.Outputs), len(outputs))
+	}
+
+	for i := range outputs {
+		if packet.Outputs[i].Unknowns != nil &&
+			len(packet.Outputs[i].Unknowns) != 0 {
+
+			return fmt.Errorf("partial output %d already has "+
```

Comment:
This might already be the case if we encoded the single or double tweak (depending on when/where this would be called).

Reply:
I've removed the error condition and instead used append, in case there's already data there.